src/ai.js

```diff
@@ -1,16 +1,651 @@
+            }
+
+            let score = this._evaluate(simulation);
+            if (score <= best_card[1]) return;
+
+            // This card is now the best card
+            best_card = [card, score];
+        });
+
+        // FIXME: Is this necessary after the last restore game in the loop above?
+        this._restoreGame();
+
+        let score = best_card[1];
+        best_card = best_card[0];
+
+        // If a card wasn't chosen, choose the first card.
+        if (!best_card) {
+            // As a backup, do this process all again but this time we don't care about the cost of cards.
+            // TODO: I'm not sure about this one. This looks like a nightmare on performance.
+            if (care_about_mana) return this.discover(cards, false);
+
+            // Choose the first discover card as the last resort.
+            this.history.push([history_name, null]);
+            return cards[0];
+        }
+
+        this.history.push([history_name, [best_card.name, score]]);
+
+        return best_card;
+    }
+
+    /**
+     * Evaluates the game
+     * 
+     * @param {Game} simulation
+     * @param {boolean} [sentiment=true] If it should perform sentiment analysis on the card's desc.
+     * 
+     * @returns {number} The score
+     */
+    _evaluate(simulation, sentiment = true) {
+        // TODO: Make this better
+        let score = 0;
+        const VALUE_BIAS = 0.1;
+
+        simulation.board.forEach(c => {
+            c.forEach(c => {
+                let bias = (c.plr.id == this.plr.id) ? 1 : -1;
+                let s = this._evaluateCard(c, sentiment);
+                //if (bias == 1 && s < 0) s = 0;
+                //if (bias == -1 && s > 0) s = -0;
+
+                score += s * bias;
+            });
+        });
+
+        [simulation.player1, simulation.player2].forEach(p => {
+            Object.entries(p).forEach(e => {
+                let [key, val] = e;
+                if (typeof val !== "number") return;
+                if (val == 0) return;
+                if (["id"].includes(key)) return;
+
+                if (["fatigue", "heroPowerCost", "overload"].includes(key)) val = -val;
+
+                let bias = 1;
+                if (p.id != this.plr.id) bias = -1;
+
+                score += val * bias * VALUE_BIAS;
+            });
+        });
+
+        [simulation.player1, simulation.player2].forEach(p => {
+            [p.deck.length, p.hand.length, p.quests.length, p.sidequests.length, p.secrets.length].forEach(val => {
+                let bias = 1;
+                if (p.id != this.plr.id) bias = -1;
+
+                score += val * bias * VALUE_BIAS;
+            });
+        });
+
+        return score;
+    }
+
+    /**
+     * Evaluates a card
+     * 
+     * @param {Card} c The card
+     * @param {boolean} [sentiment=true] If it should perform sentiment analysis on the card's description
+     * 
+     * @returns {number} The score
+     */
+    _evaluateCard(c, sentiment = true) {
```

Comment:
Maybe don't use this? Or at least put this function (and some others) in a common AI class that both ai's can inherit from.